### DISCUSSION TITLE
generateStaticParams giving 404 error on newly created jobs

### DISCUSSION THREAD

**Discussion by codesbyhusnain** [vercel/next.js#65981] [App Router]
I am using app router. I have many dynamic routes such as jobs and contracts. I am using generateStaticParams to generate my dynamic routes when I build my project.
It does work for the jobs I already created and routes to them perfectly. However, when I create a new job and try to route to it, it gives me a 404 error. 

My code:

```
import { collection, getDocs, doc, getDoc } from "firebase/firestore";
import { database } from "@/firebase";
import JobDetailsScreen from "./jobdetails";

export async function generateStaticParams() {
  const jobsCollection = collection(database, "projects");
  const jobsSnapshot = await getDocs(jobsCollection);
  const jobs = jobsSnapshot.docs.map((doc) => ({ id: doc.id }));

  return jobs.map((job) => ({
    slug: job.id,
  }));
}

// Page Component
export default async function JobDetailsPage({
  params,
}: {
  params: { slug: string };
}) {
  const jobId = params.slug;

  // Fetch job details
  const jobRef = doc(database, "projects", jobId);
  const jobSnap = await getDoc(jobRef);

  if (!jobSnap.exists()) {
    // Handle the case where job doesn't exist (e.g., show a 404 page)
    return <div>Job not found</div>;
  }

  const jobData = jobSnap.data();

  return <JobDetailsScreen jobId={params.slug} />;
}
```

**Comment by icyJoseph** [comment]
Very odd... do you maybe have https://nextjs.org/docs/app/api-reference/file-conventions/route-segment-config#dynamicparams set to false? or is this app being exported?

**Reply by codesbyhusnain** [reply]
Where do I have to set that? Because I haven't manually set it to anything. Do I have to set it to true at every dynamic page.tsx?

**Reply by codesbyhusnain** [reply]
```
// next.config.js
module.exports = {
  reactStrictMode: true,
  trailingSlash: true,
  swcMinify: true,
  output: "export",
  images: { unoptimized: true },
  experimental: {
    appDir: true,
  },
};

```

This is my next config just for reference. I do have `export` as my output but I need that in order to deploy it to firebase. If that is causing the issue what would be the alternative for that?

**Reply by icyJoseph** [reply]
Ah yeah, export is not working very well right now I think... and in this case I don't think it ever will?

The problem being that export does a Static HTML output, which means there's no server, and without a server, there's no entity to run your server side code, aka, no one to reach out to firebase.

**Reply by codesbyhusnain** [reply]
So if I remove it, can I still deploy to firebase? Should I set export to `out`?

**Reply by icyJoseph** [reply]
Depends, can your deploy environment run a Next.js app? This docs entry, https://firebase.google.com/docs/hosting/frameworks/nextjs, would seem to suggest that up to 13.4.7, is supported as a Next.js server app.

**Reply by codesbyhusnain** [reply]
Yeah but I am using 14+, app router is supposed to work similarly as pages but I think it is lacking in dynamic routing

**Reply by codesbyhusnain** [reply]
Any idea on how I can deploy a dynamic next app on firebase without `export`?
